core/src/main/java/org/apache/iceberg/BaseRewriteManifests.java

```diff
@@ -242,7 +241,7 @@ private void performRewrite(List<ManifestFile> currentManifests) {
           .executeWith(workerPool())
           .run(
               manifest -> {
-                if (predicate != null && !predicate.test(manifest)) {
+                if (containsDeletes(manifest) || !matchesPredicate(manifest)) {
```

Comment:
Our `BaseRewriteManifests` can rewrite data manifests as well as simply replace manifests rewritten by an external process. Adding support for rewriting delete manifests is out of scope. The primary and much more scalable way of rewriting manifests is via an action. This PR focuses purely on the ability to replace delete manifests rewritten externally.